src/realm-settings/NewClientProfileForm.tsx

```diff
@@ -14,15 +18,83 @@ import { FormAccess } from "../components/form-access/FormAccess";
 import { ViewHeader } from "../components/view-header/ViewHeader";
 import { Link } from "react-router-dom";
 import { useRealm } from "../context/realm-context/RealmContext";
+import { useAlerts } from "../components/alert/Alerts";
+import { useAdminClient, useFetch } from "../context/auth/AdminClient";
+import type ClientProfileRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientProfileRepresentation";
+import type ClientPolicyExecutorRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientPolicyExecutorRepresentation";
+import { HelpItem } from "../components/help-enabler/HelpItem";
+import { PlusCircleIcon } from "@patternfly/react-icons";
+import "./RealmSettingsSection.css";
+
+type NewClientProfileForm = {
+  name: string;
+  executors: ClientPolicyExecutorRepresentation[];
+  description: string;
+};
+
+const defaultValues: NewClientProfileForm = {
+  name: "",
+  executors: [],
+  description: "",
+};
 
 export const NewClientProfileForm = () => {
   const { t } = useTranslation("realm-settings");
-  const { register, errors } = useForm();
+  const { getValues, register, errors } = useForm<NewClientProfileForm>({
+    defaultValues,
+  });
   const { realm } = useRealm();
+  const { addAlert, addError } = useAlerts();
+  const adminClient = useAdminClient();
+  const [globalProfiles, setGlobalProfiles] = useState<
+    ClientProfileRepresentation[]
+  >([]);
+  const [profiles, setProfiles] = useState<ClientProfileRepresentation[]>([]);
+  const [showAddExecutorsForm, setShowAddExecutorsForm] = useState(false);
+
+  useFetch(
+    async () =>
+      await adminClient.clientPolicies.listProfiles({
+        includeGlobalProfiles: true,
+      }),
+    (profiles) => {
+      setGlobalProfiles(
+        profiles.globalProfiles as ClientProfileRepresentation[]
+      );
+      setProfiles(profiles.profiles as ClientProfileRepresentation[]);
+    },
+    []
+  );
+
+  const save = async () => {
+    const form = getValues();
+
+    const createdProfile = {
+      name: form.name,
+      executors: [],
+      description: form.description,
+    };
```

Comment:
Since the values of the form match the shape of `createdProfile` exactly you could also spread it and only overwrite the fields you want to clear.

```suggestion
    const createdProfile = {
      ...form,
      executors: [],
    };
```

Reply:
+1